zuul-core/build.gradle

```diff
@@ -14,12 +14,14 @@ dependencies {
     api "com.netflix.spectator:spectator-api:0.123.1"
     api "com.netflix.netflix-commons:netflix-commons-util:0.3.0"
 
+    api project(":zuul-discovery")
```

Comment:
If you want the dependency restriction to work (i.e. not be transitive), this should be `implementation` rather than `api`.

Reply:
the ribbon deps in `zuul-discovery` are `implementation`. That effectively means anything pulling in `zuul-discovery` won't have access to those types, which is what we want.

We can probably make this itself restrictive, but that'll apply to the new types defined in discovery, not in ribbon.